pandas/indexes/base.py

```diff
@@ -1328,22 +1328,43 @@ def _ensure_compat_concat(indexes):
 
         return indexes
 
-    def take(self, indices, axis=0, allow_fill=True, fill_value=None):
-        """
-        return a new Index of the values selected by the indexer
+    _index_shared_docs['take'] = """
+        return a new Index of the values selected by the indices
 
         For internal compatibility with numpy arrays.
 
-        # filling must always be None/nan here
-        # but is passed thru internally
+        Parameters
+        ----------
+        indices : list
+            Indices to be taken
+        axis : int, optional
+            The axis over which to select values, always 0.
+        allow_fill : bool, default True
+        fill_value : bool, default None
+            If allow_fill=True and fill_value is not None, indices specified by
+            -1 is regarded as NA. If Index doesn't hold NA, raise ValueError
 
         See also
         --------
         numpy.ndarray.take
         """
+    @Appender(_index_shared_docs['take'])
+    def take(self, indices, axis=0, allow_fill=True, fill_value=None):
 
         indices = com._ensure_platform_int(indices)
         taken = self.values.take(indices)
+
+        if self._can_hold_na:
+            # only fill if we are passing a non-None fill_value
+            if allow_fill and fill_value is not None:
```

Comment:
Actually I think we need to either make both this, and ``tseries/base`` take:

either: raise on any negative indexers except -1
or: make a new method to do this, maybe ``.take_with_fill`` and restore ``.take`` for compat with 'regular' take. These are serving double duty here.